stan/math/gpu/err/check_matrix_gpu.hpp

```diff
@@ -0,0 +1,184 @@
+#ifndef STAN_MATH_PRIM_MAT_ERR_CHECK_GPU_HPP
```

Comment:
please fix the header guard to line up with the folder structure.

Reply:
fixed in all separate files